udata/tests/__init__.py

```diff
@@ -1,83 +1,65 @@
 import unittest
 
 import pytest
+from werkzeug import Response
 
 from udata import settings
+from udata.app import UDataApp, create_app
+from udata.tests.plugin import TestClient
 
 from . import helpers
 
 
-class TestCase(unittest.TestCase):
-    settings = settings.Testing
+class TestCaseMixin:
+    app: UDataApp
 
-    @pytest.fixture(autouse=True)
-    def inject_app(self, app):
-        self.app = app
-        return self.create_app()
-
-    def create_app(self):
+    def get_settings(self, request):
         """
-        Here for compatibility legacy test classes
+        This seems really complicated for what it's doing. I think we want to create
+        an app with the default setting being `settings.Testing` and some overrides
+        from the "options" markers. But to do that `settings.Testing` should inherit from
+        `settings.Default`?
```

Comment:
I also find this logic confusing and I'm unsure how it works at the moment.
I don't think `settings.Testing` should inherit from `settings.Default`, as it is passed as override to the default ones.